### PR TITLE
Add new keyboard rev_0101, rev_0111, rev_0112 and rev_0113 for keychron.

### PR DESCRIPTION
**Pull request by lalalademaxiya1**
## Description
- Add new keyboard rev_0101, rev_0111, rev_0112 and rev_0113 for keychron.

### CODE REVIEW

**Review by noroadsleft — COMMENTED** [review]
I need to make some changes for these on the `qmk_firmware` side before I can bring these in. All of these boards have issues with their QMK Configurator implementations.

**Comment by yanfali** [comment]
@noroadsleft bump

File: public/keymaps/k/keychron_q2_rev_0112_default.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    ],
+    [
+      "KC_GRV",  "KC_BRID",  "KC_BRIU", "LGUI(KC_TAB)", "LGUI(KC_E)", "RGB_VAD", "RGB_VAI", "KC_MPRV", "KC_MPLY", "KC_MNXT", "KC_MUTE", "KC_VOLD", "KC_VOLU", "KC_TRNS",            "KC_TRNS",
+      "RGB_TOG", "RGB_MOD",  "RGB_VAI", "RGB_HUI",      "RGB_SAI",    "RGB_SPI", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS",                       "KC_TRNS",
+      "KC_TRNS", "RGB_RMOD", "RGB_VAD", "RGB_HUD",      "RGB_SAD",    "RGB_SPD", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS",            "KC_TRNS",
+      "KC_TRNS", "KC_TRNS",  "KC_TRNS", "KC_TRNS",      "KC_TRNS",    "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS",            "KC_TRNS", "KC_TRNS",
+      "KC_TRNS", "KC_TRNS",  "KC_TRNS",                                          "KC_TRNS",                                  "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS"
+    ],
+    [
+      "KC_WAVE", "KC_F1",    "KC_F2",   "KC_F3",   "KC_F4",   "KC_F5",   "KC_F6",   "KC_F7",   "KC_F8",   "KC_F9",   "KC_F10",  "KC_F11",  "KC_F12",  "KC_TRNS",            "KC_TRNS",
```
Comment (noroadsleft) [inline public/keymaps/k/keychron_q2_rev_0112_default.json:36]:
```suggestion
      "S(KC_GRV)", "KC_F1",    "KC_F2",   "KC_F3",   "KC_F4",   "KC_F5",   "KC_F6",   "KC_F7",   "KC_F8",   "KC_F9",   "KC_F10",  "KC_F11",  "KC_F12",  "KC_TRNS",            "KC_TRNS",
```

File: public/keymaps/k/keychron_q2_rev_0113_default.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    ],
+    [
+      "KC_GRV",  "KC_BRID",  "KC_BRIU", "LGUI(KC_TAB)", "LGUI(KC_E)", "RGB_VAD", "RGB_VAI", "KC_MPRV", "KC_MPLY", "KC_MNXT", "KC_MUTE", "KC_VOLD", "KC_VOLU", "KC_TRNS",            "KC_TRNS",
+      "RGB_TOG", "RGB_MOD",  "RGB_VAI", "RGB_HUI",      "RGB_SAI",    "RGB_SPI", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS",                       "KC_TRNS",
+      "KC_TRNS", "RGB_RMOD", "RGB_VAD", "RGB_HUD",      "RGB_SAD",    "RGB_SPD", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS",            "KC_TRNS",
+      "KC_TRNS", "KC_TRNS",  "KC_TRNS", "KC_TRNS",      "KC_TRNS",    "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS",            "KC_TRNS", "KC_TRNS",
+      "KC_TRNS", "KC_TRNS",  "KC_TRNS",                                          "KC_TRNS",                                  "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS"
+    ],
+    [
+      "KC_WAVE", "KC_F1",    "KC_F2",   "KC_F3",   "KC_F4",   "KC_F5",   "KC_F6",   "KC_F7",   "KC_F8",   "KC_F9",   "KC_F10",  "KC_F11",  "KC_F12",  "KC_TRNS",            "KC_TRNS",
```
Comment (noroadsleft) [inline public/keymaps/k/keychron_q2_rev_0113_default.json:36]:
```suggestion
      "S(KC_GRV)", "KC_F1",    "KC_F2",   "KC_F3",   "KC_F4",   "KC_F5",   "KC_F6",   "KC_F7",   "KC_F8",   "KC_F9",   "KC_F10",  "KC_F11",  "KC_F12",  "KC_TRNS",            "KC_TRNS",
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: public/keymaps/k/keychron_q2_rev_0112_default.json (hunk 1 above)
Comment (lalalademaxiya1) [reply public/keymaps/k/keychron_q2_rev_0112_default.json]: Thanks so much.💕

**Comment by noroadsleft** [comment]
I didn't realize until late last night that this PR had merge conflicts, which can't be resolved on the web interface. Will merge PR No. 1090 instead.

**Pull request closed without merging**